### ISSUE TITLE
Multiple result builders in async adapter

### ISSUE THREAD

**Issue by elijahbenizzy** [DAGWorks-Inc/hamilton#1022]
See Slack conversation: https://hamilton-opensource.slack.com/archives/C03M33QB4M8/p1720718536565439

Ryan Whitten

  1 hour ago
Hey all, I'm upgrading to the new AsyncBuilder/AsyncDriver and wondering if I'm correctly setting the result_builder. With the regular Builder I would normally set it using with_adapters() , e.g.:
```python
result_builder = base.PandasDataFrameResult()
await AsyncBuilder().with_config(self.config).with_modules(*self.included_modules).with_adapters(result_builder).build()
but this raises ValueError("You cannot pass more than one result builder to the async driver. Please pass in a single result builder") here: https://github.com/DAGWorks-Inc/hamilton/blob/main/hamilton/async_driver.py#L225-L229
```
Looks like that is due to a default DictResult getting set if the AsyncBuilder.result_builder is None: https://github.com/DAGWorks-Inc/hamilton/blob/main/hamilton/async_driver.py#L478-L480
I'm able to work around it by setting the builder's result_builder attr before calling build(), but not sure if that is the intended usage:
```python
result_builder = base.PandasDataFrameResult()
builder = AsyncBuilder().with_config(self.config).with_modules(*self.included_modules)
builder.result_builder = result_builder
await builder.build()
```

[async_driver.py](https://github.com/DAGWorks-Inc/hamilton/blob/main/hamilton/async_driver.py)
<https://github.com/[DAGWorks-Inc/hamilton](https://github.com/DAGWorks-Inc/hamilton)|DAGWorks-Inc/hamilton>DAGWorks-Inc/hamilton | Added by [GitHub](https://hamilton-opensource.slack.com/services/B03M52ALBGB)

[async_driver.py](https://github.com/DAGWorks-Inc/hamilton/blob/main/hamilton/async_driver.py)
<https://github.com/[DAGWorks-Inc/hamilton](https://github.com/DAGWorks-Inc/hamilton)|DAGWorks-Inc/hamilton>DAGWorks-Inc/hamilton | Added by [GitHub](https://hamilton-opensource.slack.com/services/B03M52ALBGB)

**Comment by skrawcz** [comment]
We can close this once we push a release right?

**Comment by elijahbenizzy** [comment]
This has been fixed in 1.71.0!
